SSL_R_SHORT_READ doesn't exist in OpenSSL 1.1

# 環境 Environment

* OS version: OS: Arch Linux, Kernel: x86_64 Linux 5.1.2-arch1-1-ARCH

* Elona foobar verision: develop branch

# 詳細 Details

```
[ 13%] Building CXX object src/spider/CMakeFiles/spider_boost_asio.dir/http/backends/boost_asio/request.cpp.o
/home/vorlent/Projects/ElonaFoobar/src/spider/http/backends/boost_asio/request.cpp: In member function ‘void elona::spider::http::{anonymous}::Session::on_shutdown(const error_code&)’:
/home/vorlent/Projects/ElonaFoobar/src/spider/http/backends/boost_asio/request.cpp:389:48: error: ‘SSL_R_SHORT_READ’ was not declared in this scope
                  ERR_GET_REASON(ec.value()) == SSL_R_SHORT_READ))
                                                ^~~~~~~~~~~~~~~~
/home/vorlent/Projects/ElonaFoobar/src/spider/http/backends/boost_asio/request.cpp:389:48: note: suggested alternative: ‘SSL_F_SSL_READ’
                  ERR_GET_REASON(ec.value()) == SSL_R_SHORT_READ))
                                                ^~~~~~~~~~~~~~~~
                                                SSL_F_SSL_READ

```

# 再現方法 How to reproduce

make build with openssl 1.1 (default on Arch Linux)

--- a/src/spider/http/backends/boost_asio/request.cpp
+++ b/src/spider/http/backends/boost_asio/request.cpp
@@ -386,7 +386,8 @@ private:
         {
             if (ec == asio::error::eof ||
                 (ec.category() == asio::error::get_ssl_category() &&
-                 ERR_GET_REASON(ec.value()) == SSL_R_SHORT_READ))
+                 ERR_GET_REASON(ec.value()) ==
+                     asio::ssl::error::stream_errors::stream_truncated))
             {
                 // This error can be ignored.
                 // http://stackoverflow.com/questions/25587403/boost-asio-ssl-async-shutdown-always-finishes-with-an-error

